src/decoder/mod.rs

```diff
@@ -699,20 +701,20 @@ impl<R: Read + Seek> Decoder<R> {
         }
 
         self.chunk_type =
+            // TODO: This needlessly allocates a vector and then discards it. Maybe init decoders here.
             match (
-                self.get_tag_u32(Tag::RowsPerStrip),
-                self.get_tag_u32(Tag::TileWidth),
-                self.get_tag_u32(Tag::TileLength),
+                self.get_tag_u64_vec(Tag::StripByteCounts),
+                self.get_tag_u64_vec(Tag::StripOffsets),
+                self.get_tag_u64_vec(Tag::TileByteCounts),
+                self.get_tag_u64_vec(Tag::TileOffsets),
```

Comment:
Couldn't you just do `self.ifd.as_ref().unwrap().contains_key(&tag)` instead for each of these? That would avoid the read and the vector allocation.

Reply:
Thank  you, that's  much better... I feel kinda dumb, because I spent some time looking for something like contains_key(), but didn't realize that Directory is a HashMap.